plugins/vendor/chargeBee/services/ChargeBeeVendorService.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * @service chargeBeeVendor
+ * @package plugins.chargeBee
+ * @subpackage api.services
+ */
+class ChargeBeeVendorService extends KalturaBaseService
+{
+	const MAP_NAME = 'vendor';
+	const CONFIGURATION_PARAM_NAME = 'chargeBee';
+	
+	protected static $PARTNER_NOT_REQUIRED_ACTIONS = array('handleNotification');
+	
+	/**
+	 * no partner will be provided by vendors as this called externally and not from kaltura
+	 * @param string $actionName
+	 * @return bool
+	 */
+	protected function partnerRequired($actionName)
+	{
+		return in_array ($actionName, self::$PARTNER_NOT_REQUIRED_ACTIONS);
+	}
+
+	/**
+	 * Add chargeBee vendor integration object
+	 *
+	 * @action add
+	 * @param KalturaChargeBeeVendorIntegration $chargeBeeVendorIntegration
+	 * @return KalturaChargeBeeVendorIntegration
+	 */
+	public function addAction(KalturaChargeBeeVendorIntegration $chargeBeeVendorIntegration)
+	{
+		$dbChargeBeeVendorIntegration = $chargeBeeVendorIntegration->toInsertableObject();
+		/* @var $dbChargeBeeVendorIntegration kChargeBeeVendorIntegration */
+
+		$dbChargeBeeVendorIntegration->setAccountId($chargeBeeVendorIntegration->subscriptionId);
+		$dbChargeBeeVendorIntegration->setVendorType($chargeBeeVendorIntegration->type);
+		$dbChargeBeeVendorIntegration->setPartnerId($chargeBeeVendorIntegration->partnerId);
+		$dbChargeBeeVendorIntegration->setStatus(VendorStatus::ACTIVE);
```

Comment:
Since only system partners can make this call, lets take the partnerId from the impersonated ks